scripts/tests/local.py

```diff
@@ -0,0 +1,786 @@
+    show_default=True,
+    help="At the end of the execution, show how many seconds each test took.",
+)
+@click.option(
+    "--runner",
+    default="none",
+    type=click.Choice(list(__RUNNERS__.keys()), case_sensitive=False),
+    help="Determines the verbosity of script output",
+)
+def python_tests(
+    test_filter, from_filter, from_skip_filter, dry_run, show_timings, runner
+):
+    """
+    Run python tests via `run_python_test.py`
+
+    Constructs the run yaml in `out/test_env.yaml`. Assumes that binaries
+    were built already, generally with `build` (or separate `build-python` and `build-apps`).
+    """
+    runner = __RUNNERS__[runner]
+
+    def as_runner(path):
+        return _maybe_with_runner(os.path.basename(path), path, runner)
+
+    # create an env file
+    target_prefix = _get_native_machine_target()
+    with open("out/test_env.yaml", "wt") as f:
+        f.write(
+            textwrap.dedent(
+                f"""\
+            ALL_CLUSTERS_APP: {as_runner(f'out/{target_prefix}-all-clusters-no-ble-clang-boringssl/chip-all-clusters-app')}
+            CHIP_LOCK_APP: {as_runner(f'out/{target_prefix}-lock-no-ble-clang-boringssl/chip-lock-app')}
+            ENERGY_MANAGEMENT_APP: {
+                as_runner(f'out/{target_prefix}-energy-management-no-ble-clang-boringssl/chip-energy-management-app')}
+            LIT_ICD_APP: {as_runner(f'out/{target_prefix}-lit-icd-no-ble-clang-boringssl/lit-icd-app')}
+            CHIP_MICROWAVE_OVEN_APP: {
+                as_runner(f'out/{target_prefix}-microwave-oven-no-ble-clang-boringssl/chip-microwave-oven-app')}
+            CHIP_RVC_APP: {as_runner(f'out/{target_prefix}-rvc-no-ble-clang-boringssl/chip-rvc-app')}
+            NETWORK_MANAGEMENT_APP: {
+                as_runner(f'out/{target_prefix}-network-manager-ipv6only-no-ble-clang-boringssl/matter-network-manager-app')}
+            TRACE_APP: out/trace_data/app-{{SCRIPT_BASE_NAME}}
+            TRACE_TEST_JSON: out/trace_data/test-{{SCRIPT_BASE_NAME}}
+            TRACE_TEST_PERFETTO: out/trace_data/test-{{SCRIPT_BASE_NAME}}
+            """
+            )
+        )
+
+    if not test_filter.startswith("*"):
+        test_filter = "*" + test_filter
+    if not test_filter.endswith("*"):
+        test_filter = test_filter + "*"
+
+    if from_filter:
+        if not from_filter.startswith("*"):
+            from_filter = "*" + from_filter
+        if not from_filter.endswith("*"):
+            from_filter = from_filter + "*"
+
+    if from_skip_filter:
+        if not from_skip_filter.startswith("*"):
+            from_skip_filter = "*" + from_skip_filter
+        if not from_skip_filter.endswith("*"):
+            from_skip_filter = from_skip_filter + "*"
+
+    # This MUST be available or perfetto dies. This is VERY annoying to debug
+    if not os.path.exists("out/trace_data"):
+        os.mkdir("out/trace_data")
+
+    # IGNORES are taken out of `src/python_testing/execute_python_tests.py` in the SDK
```

Comment:
Can we have a single source of truth for this?  Otherwise, these lists will diverge, which will cause problems.

Reply:
will work towards it with @vatsalghelani-csa - we probably want something loadable and then we can use it from both places (one our official runner and one being this local runner)